luarules/gadgets/unit_minesweeper_detection.lua

```diff
@@ -5,201 +5,115 @@ function gadget:GetInfo()
 	end
 end
 
 function gadget:UnitDestroyed(unitID, unitDefID, unitTeam)
-	if minesweepers[unitTeam][unitID] then
+	if minesweeperRanges[unitDefID] and minesweepers[unitTeam][unitID] then
 		minesweepers[unitTeam][unitID] = nil
 	end
 end
 
-function gadget:GameFrame(f)
+local function processesTeamsSweepers(currentTeam)
 	local minesToReveal = {}
-	--generate list of active teams. this is list of mines that team can see
-
-
-	for teamouter in pairs(teamIDs) do
-
-		--if this team's update cycle
-		if (f % 37 + (teamouter*2)) == 1 then
-			for teamprime in pairs(teamIDs) do
-				minesToReveal[teamprime] = {}
-			end
-			
-			--minesToReveal[teamouter] = {}
-
-			--Spring.Echo('hornet minesweepers')
-			--Spring.Debug.TableEcho(minesweepers)
-			--find all minesweepers
-			for miner, minerdef in next, (minesweepers[teamouter] or {}) do
-
-				--Spring.Echo('hornet miner', miner)
-				--Spring.Echo('hornet minerdef', minerdef)
-
-				
-				local minerTeam = Spring.GetUnitTeam(miner)
-				local x, y, z = Spring.GetUnitPosition(miner)
-
-				if x ~= nil and z ~= nil then
-					--bypass 'impossible' bug causing logspam 'sometimes'
-					
-
-
-					local nearUnits = Spring.GetUnitsInCylinder(x, z, minesweeperRanges[minerdef])
-
-					--Spring.Echo('hornet minerTeam', minerTeam)
-					--Spring.Echo('hornet minerdef minesweeperRanges[minerdef]', minerdef, minesweeperRanges[minerdef])
-
-					--Spring.Echo('hornet nearUnits')
-					--Spring.Debug.TableEcho(nearUnits)
-					for _, nearUnit in ipairs(nearUnits) do
-						--Spring.Echo('hornet poi Spring.GetUnitTeam(nearUnit)', Spring.GetUnitTeam(nearUnit))
-						--local mineTeam = Spring.GetUnitTeam(nearUnit)
-						if minerTeam ~= Spring.GetUnitTeam(nearUnit) then
-							local nearUnitDefID = Spring.GetUnitDefID(nearUnit)
-						
-							if mineIDs[nearUnitDefID] then
-								--Spring.Echo('hornet adding  minesToReveal[minerTeam][nearUnit] = nearUnit', minerTeam, nearUnit)
-								minesToReveal[minerTeam][nearUnit] = nearUnit
-							end
-							
-						end
 
+	for sweeper, sweeperDef in pairs(minesweepers[currentTeam])  do
+		local x, y, z = Spring.GetUnitPosition(sweeper)
+		if x and z then
+			local nearUnits = Spring.GetUnitsInCylinder(x, z, minesweeperRanges[sweeperDef])
+			for _, nearUnit in ipairs(nearUnits) do
+				local nearUnitDefID = Spring.GetUnitDefID(nearUnit)
+				if currentTeam ~= Spring.GetUnitTeam(nearUnit) then
+					if mineIDs[nearUnitDefID] then
+						minesToReveal[nearUnit] = true
 					end
-				else
-					minesweepers[teamouter][miner] = nil
-					--unset , errant entry somehow
-
 				end
 			end
+		end
+	end
 
+	for mineToReveal, _ in pairs(minesToReveal) do
+		if revealedMines[currentTeam][mineToReveal] == nil then
+			revealedMines[currentTeam][mineToReveal] = true
+			--show in vision and radar, and prevent engine from immediately resseting that state
+			Spring.SetUnitLosState(mineToReveal, currentTeam, 3)
+			Spring.SetUnitLosMask(mineToReveal, currentTeam, 3)
```

Comment:
Not touching the lines that actually crashed?

Reply:
i haven't figured out why those crashed it at the time or crying over this

Reply:
switched to use allied teams